OPHD/UI/ProductListBox.cpp

```diff
@@ -26,15 +26,19 @@ void ProductListBox::productPool(ProductPool& pool)
 {
 	clear();
 
-	for (std::size_t product_type = 0; product_type < static_cast<std::size_t>(ProductType::PRODUCT_COUNT); ++product_type)
+	const auto totalProducts = static_cast<std::size_t>(ProductType::PRODUCT_COUNT);
+
+	for (std::size_t product_type = 0; product_type < totalProducts; ++product_type)
 	{
-		const auto productCount = pool.count(static_cast<ProductType>(product_type));
+		const auto productType = static_cast<ProductType>(product_type);
+		const auto productCount = pool.count(productType);
```

Comment:
Variable names could probably be improved here.

I find `totalProducts` to be a bit misleading of a name. It's basically the number of product types. Maybe `productTypeCount`?

There is both a `product_type` and a `productType`. I would be tempted to rename `product_type` to `productTypeIndex`.

Reply:
I agree. I didn't want to spend too much time on it but I noticed that naming was a little weird here.